services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -322,15 +320,8 @@ private void onTick(Date date) {
 
   @Subscribe
   public void setNodeSlotAccordingToDBStore(Store store) {
-    Checkpoint finalizedCheckpoint = store.getFinalizedCheckpoint();
-    BeaconState state = store.getBlockState(finalizedCheckpoint.getRoot());
-    UnsignedLong unixTimeStamp = UnsignedLong.valueOf(Instant.now().getEpochSecond());
-    UnsignedLong genesisTime = state.getGenesis_time();
-    chainStorageClient.setGenesisTime(genesisTime);
-    nodeSlot = unixTimeStamp.minus(genesisTime).dividedBy(UnsignedLong.valueOf(SECONDS_PER_SLOT));
-    STDOUT.log(
-        Level.INFO,
-        "Database Store genesis time " + genesisTime + " and current node slot " + nodeSlot + ".",
-        ALogger.Color.GREEN);
+    Bytes32 headBlockRoot = get_head(store);
+    chainStorageClient.initializeFromStore(store, headBlockRoot);
+    STDOUT.log(Level.INFO, "Node being started from database.", ALogger.Color.GREEN);
```

Comment:
We lost setting the `nodeSlot` in this move but need it so that we start from the current slot instead of from 0 all the time.

Reply:
onStoreInitializedEvent handles the nodeSlot anyway, so I think the previous setting of nodeSlot was redundant: https://github.com/PegaSysEng/artemis/blob/master/services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java#L265-L265

Reply:
Ah yes, right you are.